### ISSUE TITLE
Bug: [consistent-type-imports] sometimes does not report mixes of modules and types in one import as errors.

### ISSUE THREAD

**Issue by kylemh** [typescript-eslint/typescript-eslint#11096]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=JYWwDg9gTgLgBAbzgVwM4FMCSATANHGATzHTgGEJwIA7damABSgjFXyJLgCV0BDAYxgA5CNlIBfOADNmIOAHIofQfIDcAKE0dSAFXSp4AXnKVItekxaoAPPLDyAfBu1w9BgExxjPAcNHoNfhpUCAAbdAA6UIgAcwAKJDQsbDhxAEo4XlQUagBraggAd2pM7Ld4LJz8opLK8s9K3mpCDXUgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMkRNGhTI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import { useId, type ComponentProps, type ReactNode } from 'react';
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-type-imports": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

// expected repro to error and auto-fix to
import { useId } from 'react';
import type { ComponentProps, ReactNode } from 'react';

### Actual Result

No error. "False-positive", I think it's called?

**Comment by bradzacher** [comment]
The rule does not enforce using top level vs inline specifiers. The option merely controls how the fixer behaves. If you want to enforce usage of specific specifiers then check the related rules.

https://typescript-eslint.io/rules/consistent-type-imports/#related-to
